apps/cowswap-frontend/src/modules/injectedWidget/updaters/CowEventsUpdater.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react'
+
+import { CowEventListener, CowEventListeners, CowEvents } from '@cowprotocol/events'
+
+import { useCowEventEmitter } from 'common/hooks/useCowEventEmitter'
+
+import { COW_SWAP_WIDGET_EVENT_KEY } from '../consts'
+
+const TARGET_ORIGIN = '*' // Change to CoW specific origin in production
+
+const ALL_EVENTS = Object.values(CowEvents)
+
+export function CowEventsUpdater() {
+  const cowEventEmitter = useCowEventEmitter()
+
+  // Setup listeners only once
+  useEffect(() => {
+    // cowEventEmitter.on(CowEvents)
+    const allHandlers: CowEventListeners = ALL_EVENTS.map((event) => {
+      return {
+        event,
+        handler: (payload: any) => forwardEventToIframe(event, payload),
+      }
+    })
+
+    allHandlers.forEach((listener) => cowEventEmitter.on(listener as CowEventListener<CowEvents>))
+
+    // Cleanup: Remove all listeners
+    return () => {
+      allHandlers.forEach((listener) => cowEventEmitter.off(listener as CowEventListener<CowEvents>))
+    }
+  }, [])
```

Comment:
Lint error here.
If you use `cowEventEmitter` as a global const the error will disappear

Reply:
The lint issue was corrected at some point later https://github.com/cowprotocol/cowswap/blob/a6026f029c1cf5795ea0ff55d58fb5804ee0e1ee/apps/cowswap-frontend/src/modules/injectedWidget/updaters/CowEventsUpdater.ts#L34

Reply:
I thought a bit more about this, you convinced me. I will kill the context, and leave a single eventEmitter. Expect a PR soon!

Reply:
https://github.com/cowprotocol/cowswap/pull/3839